plugins/telemetry-otel/src/internalClusterTest/java/org/opensearch/telemetry/metrics/TelemetryMetricsEnabledSanityIT.java

```diff
@@ -133,21 +133,25 @@ public void testGauge() throws Exception {
         InMemorySingletonMetricsExporter.INSTANCE.reset();
         Tags tags = Tags.create().addTag("test", "integ-test");
         final AtomicInteger testValue = new AtomicInteger(0);
-        Supplier<Double> valueProvider = () -> { return Double.valueOf(testValue.incrementAndGet()); };
+        Supplier<Double> valueProvider = () -> Double.valueOf(testValue.incrementAndGet());
         Closeable gaugeCloseable = metricsRegistry.createGauge(metricName, "test", "ms", valueProvider, tags);
-        // Sleep for about 2.2s to wait for metrics to be published.
-        Thread.sleep(2200);
-
         InMemorySingletonMetricsExporter exporter = InMemorySingletonMetricsExporter.INSTANCE;
 
-        assertTrue(getMaxObservableGaugeValue(exporter, metricName) >= 2.0);
+        // Wait until at least 2 observations have been published avoid fixed sleeps.
+        assertBusy(() -> assertTrue(getMaxObservableGaugeValue(exporter, metricName) >= 2.0));
         gaugeCloseable.close();
-        double observableGaugeValueAfterStop = getMaxObservableGaugeValue(exporter, metricName);
-
-        // Sleep for about 1.2s to wait for metrics to see that closed observableGauge shouldn't execute the callable.
-        Thread.sleep(1200);
-        assertEquals(observableGaugeValueAfterStop, getMaxObservableGaugeValue(exporter, metricName), 0.0);
-
+        // After close, allow any in-flight collection to finish, but ensure the value eventually becomes stable.
```

Comment:
How is this going to wait for in-flight collections to finish? The code is going to invoke `getMaxObservableGaugeValue(exporter, metricName)` twice in very quick succession, and assuming it returns the same value both times it will then pass.